satpy/tests/test_readers.py

```diff
@@ -763,3 +777,19 @@ def test_viirs_override_keys(self):
         self.assertEqual(6, len(groups[0]['viirs_sdr']))
         # 5 granules * 3 file types
         self.assertEqual(5 * 3, len(groups[1]['viirs_sdr']))
+
+    def test_multi_readers(self):
+        """Test passing multiple readers."""
+        from satpy.readers import group_files
+        groups = group_files(
+                self.g16_files + self.noaa20_files,
+                reader=("abi_l1b", "viirs_sdr"))
+        assert len(groups) == 11
+        # test that they're grouped together when time threshold is huge and
+        # only time is used to group
+        groups = group_files(
+                self.g16_files + self.noaa20_files,
+                reader=("abi_l1b", "viirs_sdr"),
+                group_keys=("start_time"),
```

Comment:
This is creating a string not a tuple, right? This shouldn't work both based on the docstring and just actual behavior. Wouldn't this mean the group_keys are `('s', 't', 'a', 'r', 't', ...)`?

Reply:
You're right!  It turns out that we have:

`group_key = tuple(file_info.get(k) for k in group_keys)`

such that it becomes `(None, None, None, None, None, None, None, None, None, None)`

which in turn leads to everything being treated as one group.  So it returns a single group, but for the wrong reason.

I don't know if this is entirely desirable.  I suppose the intention is that keys not available in the file info are ignored for the purposes of grouping, but maybe if none of the keys are found a warning would be in order?  This however happens at four levels of for loops deep.  It's looping over potentially all readers, then all filetypes, then all matching filenames.  I'll add a check here.

Reply:
Now raising a `UserWarning` if none of the group keys are found.  This may be due to the user passing a string, or due to the user simply passing keys that aren't in use.  Should I add a separate explicit that the user isn't passing a string?

Reply:
I personally think the comma-separated warning message should make it pretty clear what is going on, but I don't feel very strongly about it.

Reply:
Ok, I'll leave it as it is.  This area is touched much less by new users than `sc.load("natural_color")` is so I agree that it's good enough.